src/backend/distributed/utils/node_metadata.c

```diff
@@ -87,21 +93,45 @@ master_add_node(PG_FUNCTION_ARGS)
 	bool hasMetadata = false;
 	bool nodeAlreadyExists = false;
 
-	Datum returnData = AddNodeMetadata(nodeNameString, nodePort, groupId, nodeRack,
-									   hasMetadata, &nodeAlreadyExists);
+	Datum nodeRecord = AddNodeMetadata(nodeNameString, nodePort, groupId, nodeRack,
+									   hasMetadata, NODE_STATE_INACTIVE,
+									   &nodeAlreadyExists);
 
 	/*
-	 * After adding new node, if the node is not already exist, we  replicate all existing
-	 * reference tables to the new node. ReplicateAllReferenceTablesToAllNodes replicates
-	 * reference tables to all nodes however, it skips nodes which already has healthy
-	 * placement of particular reference table.
+	 * After adding new node, if the node is not already exist, we will activate
+	 * the node. This means we will replicate all reference tables to the new
+	 * node.
 	 */
 	if (!nodeAlreadyExists)
 	{
-		ReplicateAllReferenceTablesToAllNodes();
+		nodeRecord = ActivateNode(nodeNameString, nodePort);
 	}
 
-	PG_RETURN_CSTRING(returnData);
+	PG_RETURN_CSTRING(nodeRecord);
+}
+
+
+/*
+ * master_prepare_node function adds a new node to the cluster as inactive node
```

Comment:
This name come up in the e-mail discussion of the change. I also suggested `master_add_inactive_node` and `master_add(_inactive)_node_to_metadata` but we decided not use them because we use inactive and metadata terms in different places with slightly different meanings.

master_prepare_node name come from Ozgun, but I think he does not have a strong preference either. Do you have a suggestion in this matter?

Quote from mail discussion
> Both master_add_inactive_node() and master_add_(inactive)_metadata() describe our behavior in terms of other api names: add, in-active, and metadata. Rather than use mix of words that have other meanings, we use a different word for now. Say, master_prepare_node()?